### PR TITLE
[armnn] Added tensor filter extension of armnn @open sesame 12/23 18:49

### PR DESCRIPTION
**Pull request by kparichay**
Added tensor filter extension of armnn

Related Issue: #1817 

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/meson.build [hunk 1]
```diff
@@ -30,6 +32,38 @@ if get_option('enable-nnfw-runtime')
...
+    dependencies: nnstreamer_filter_armnn_deps,
+    install: true,
+    install_dir: filter_subplugin_install_dir
+  )
+  static_library('nnstreamer_filter_armnn',
+    nnstreamer_filter_armnn_sources,
+    cpp_args: ['-Wno-sign-compare'],
+    dependencies: nnstreamer_filter_armnn_deps,
+    install: false,
+    install_dir: filter_subplugin_install_dir
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build]: let's check install directory for static libraries (filter_subplugin_install_dir : nnstreamer_libdir)

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/meson.build]: Updated.

**Comment by kparichay** [comment]
V2: Updated the static library install paths for armnn and nnfw subplugins

File: ext/nnstreamer/tensor_filter/meson.build [hunk 2]
```diff
@@ -15,21 +15,55 @@ if get_option('enable-nnfw-runtime')
...
+    cpp_args: ['-Wno-sign-compare'],
+    dependencies: nnstreamer_filter_armnn_deps,
+    install: true,
+    install_dir: filter_subplugin_install_dir
+  )
+  static_library('nnstreamer_filter_armnn',
+    nnstreamer_filter_armnn_sources,
+    cpp_args: ['-Wno-sign-compare'],
+    dependencies: nnstreamer_filter_armnn_deps,
+    install: false,
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build]: Question: (armnn and nnfw) the static libraries will not be installed, any reason for this?

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/meson.build]: updated

File: ext/nnstreamer/tensor_filter/meson.build [hunk 3]
```diff
@@ -15,21 +15,55 @@ if get_option('enable-nnfw-runtime')
...
   nnfw_plugin_lib = shared_library('nnstreamer_filter_nnfw',
     nnstreamer_filter_nnfw_sources,
+    cpp_args: ['-Wno-sign-compare'],
     dependencies: nnstreamer_filter_nnfw_deps,
     install: true,
     install_dir: filter_subplugin_install_dir
   )
   static_library('nnstreamer_filter_nnfw',
     nnstreamer_filter_nnfw_sources,
+    cpp_args: ['-Wno-sign-compare'],
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/meson.build]: How about setting this argument as the default one in the root meson.build?

File: packaging/nnstreamer.spec [hunk 4]
```diff
@@ -4,6 +4,11 @@
 %define		nnstexampledir	/usr/lib/nnstreamer/bin
 %define		tensorflow_support	0
 %define		enable_nnfw	1
+%define		armnn_support 0
+
+%ifnarch %arm aarch64
+%define armnn_support 0
```
Comment (dongju-chae) [inline packaging/nnstreamer.spec]: Then, who enables `armnn_support`?

File: ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc [hunk 5]
```diff
@@ -0,0 +1,684 @@
...
+    } else {
+      return 1;
+    }
+  }
+
+  if (prop->model_files[0] == NULL)
+    return -EINVAL;
+
+  core = new ArmNNCore (prop->model_files[0], hw);
+  if (core == NULL) {
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc]: `new` does not return NULL. it raises bad_alloc exception

File: packaging/nnstreamer.spec (hunk 4 above)
Comment (kparichay) [reply packaging/nnstreamer.spec]: Currently `armnn_support` is disabled as it is not available via OBS. Once it is available, this will be enabled.

Comment (dongju-chae) [reply packaging/nnstreamer.spec]: OK

File: ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc (hunk 5 above)
Comment (kparichay) [reply ext/nnstreamer/tensor_filter/tensor_filter_armnn.cc]: Updated. Also updated error handling in loadModel correspondingly.

File: ext/nnstreamer/tensor_filter/meson.build (hunk 3 above)
Comment (kparichay) [reply ext/nnstreamer/tensor_filter/meson.build]: Yes, added a new PR #1984 for this, as it was out of scope for this PR.

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/meson.build]: These args have removed. PR #1984 removes from other extensions as well.

File: ext/nnstreamer/tensor_filter/meson.build [hunk 6]
```diff
@@ -257,6 +287,7 @@ if get_option('enable-cppfilter')
 
   shared_library('nnstreamer_filter_cpp',
     nnstreamer_filter_cpp_sources,
+    cpp_args: ['-Wno-sign-compare'],
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build]: plz remove this line also

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/meson.build]: done.

**Pull request merged**